### PR TITLE
Add Silk Engine

### PR DESCRIPTION
**Pull request by MichaelHinrichs**
### SteamDB app page links to a few games using this
https://steamdb.info/app/216910/
https://steamdb.info/app/232750/
https://steamdb.info/app/303790/
https://steamdb.info/app/338390/
https://steamdb.info/app/1283400/

### Brief explanation of the change
Detect Silk engine

### CODE REVIEW

File: descriptions/Engine.Silk.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[**Silk**](https://spiders-games.com/silk-engine) is an engine designed and developed internally based on the PhyreEngine (trademark of Sony Computer Entertainment Europe).
+A tool kit and editors have also been developed so its developers can quickly develop high quality games for these platforms.
```
Comment (Lifeismana) [inline descriptions/Engine.Silk.md:2]:
```suggestion
[**Silk Engine**](https://spiders-games.com/silk-engine) is an engine developed by Spire, based on [PhyreEngine](/tech/Engine/Phyre/) and first used in the game [Faery - Legends of Avalon](/app/338390/) released in 2010.
```

**Comment by Lifeismana** [comment]
needs a non matching test

**Comment by Lifeismana** [comment]
https://github.com/SteamDatabase/FileDetectionRuleSets/blob/main/CONTRIBUTING.md

**Comment by MichaelHinrichs** [comment]
I don't understand how _NonmatchingTests.txt works. This isn't regex. Do you expect steamdb to find a false positive file that's actually named "notactuallysilk.dll"?

**Comment by Lifeismana** [comment]
did you take the time to read the entire link i gave you ?
> If the rule is written correctly, it should NOT match any of these filenames.
this file is made of filenames that shouldn't match any of the rules, this has no link to steamdb

**Comment by MichaelHinrichs** [comment]
Yes, i understand. i said FALSE positive. As in a file to NOT match.

**Comment by MichaelHinrichs** [comment]
Bruh.

**Comment by MichaelHinrichs** [comment]
You still never explained what Non Matching Tests are or how they work.

**Comment by Lifeismana** [comment]
They are strings that shouldn't match any of the rules, they are there to make sure there's no error in the rules

**Comment by MichaelHinrichs** [comment]
...Ooooohhhhh! You mean so that GitHub Actions stops the pull if the regex in rules.ini is too broad. I thought it was a list of what NOT to match that had priority over rules.ini, so if steam db finds something that's on the non matching text, it wouldn't count.

**Pull request merged**